src/rules/no-firestore-object-arrays.ts

```
@@ -176,8 +141,206 @@ Prefer:
+          break;
+        }
+      }
+    };
+
+    for (const stmt of (sourceCode.ast.body as unknown as TSESTree.Node[])) {
+      visitNode(stmt);
+    }
+
+    const seenAlias = new Set<string>();
+
+    const isPrimitiveLikeAlias = (name: string, depth: number = 0): boolean => {
```

ESLint

[error] Type number trivially inferred from a number literal, remove type annotation.

(@typescript-eslint/no-inferrable-types)